src/qt/pivx/settings/settingsmultisendwidget.cpp

```diff
@@ -311,9 +311,8 @@ void SettingsMultisendWidget::addMultiSend(QString address, int percentage, QStr
 
     if (walletModel && walletModel->getAddressTableModel()) {
         // update the address book with the label given or no label if none was given.
-        CBitcoinAddress address(strAddress);
         std::string userInputLabel = addressLabel.toStdString();
-        walletModel->updateAddressBookLabels(address.Get(), (userInputLabel.empty()) ? "(no label)" : userInputLabel,
+        walletModel->updateAddressBookLabels(DecodeDestination(strAddress), (userInputLabel.empty()) ? "(no label)" : userInputLabel,
```

Comment:
this:
```cpp
if (!IsValidDestinationString(strAddress)) {
    ...
}
...
if (walletModel && walletModel->getAddressTableModel()) {
    ...
    ...DecodeDestination(strAddress)...
}
```
results in a double decoding operation (for valid destinations).

Might be better to do:
```cpp
CTxDestination dest = DecodeDestination(strAddress);
if (!IsValidDestination(dest)) {
    ...
}
...
if (walletModel && walletModel->getAddressTableModel()) {
    ...
    ...dest...
}
```

Reply:
good catch, done.